### DISCUSSION TITLE
C API for reuse in other languages?

### DISCUSSION THREAD

**Discussion by ddevienne** [ratatui/ratatui#701] [Discussion]
Just wondering whether it's technically possible. And whether it might happen. Thanks, --DD

**Comment by joshka** [comment]
We don't have any plans to do this. Talking a brief look at what would be required (https://doc.rust-lang.org/nomicon/ffi.html#calling-rust-code-from-c), it seems like the changes would likely be too intrusive to make to the library itself and would add maintenance and support burden.

I'd suggest writing a small wrapper library that calls ratatui and exposes the necessary methods to your C code. I'd be keen to read a blog post about it when you're done!
